### PR TITLE
Add more checks for LC in native protocol.

### PR DESCRIPTION
**Pull request by KochetovNicolai**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### CODE REVIEW

File: src/Common/ErrorCodes.cpp [hunk 1]
```diff
@@ -566,6 +566,7 @@
     M(595, BZIP2_STREAM_ENCODER_FAILED) \
     M(596, INTERSECT_OR_EXCEPT_RESULT_STRUCTURES_MISMATCH) \
     M(597, NO_SUCH_ERROR_CODE) \
+    M(598, INVALID_BINARY_DATA) \
```
Comment (vdimir) [inline src/Common/ErrorCodes.cpp:569]: Are `INCORRECT_DATA` or `CORRUPTED_DATA` not suitable?

File: src/DataTypes/Serializations/SerializationLowCardinality.cpp [hunk 2]
```diff
@@ -702,7 +713,7 @@ void SerializationLowCardinality::deserializeBinaryBulkWithMultipleStreams(
             auto & index_type = low_cardinality_state->index_type;
             auto & global_dictionary = low_cardinality_state->global_dictionary;
 
-            index_type.deserialize(*indexes_stream);
+            index_type.deserialize(*indexes_stream, settings);
```
Comment (vdimir) [inline src/DataTypes/Serializations/SerializationLowCardinality.cpp:716]: Maybe pass only `settings.native_format` ?

File: tests/queries/0_stateless/2010_lc_native.python [hunk 3]
```diff
@@ -0,0 +1,302 @@
...
+
+def readStringBinary(s):
+    size = readVarUInt(s)
+    s = readStrict(s, size)
+    return s.decode('utf-8')
+
+
+def sendHello(s):
+    ba = bytearray()
+    writeVarUInt(0, ba) # Hello
```
Comment (vdimir) [inline tests/queries/0_stateless/2010_lc_native.python:82]: Maybe some code would be simpler using [struct](https://docs.python.org/3/library/struct.html) module, not sure

**Review by vdimir — APPROVED** [review]

**Comment by KochetovNicolai** [comment]
Integration tests (thread) - flacky

**Pull request merged**